Sources/ZcashLightClientKit/Block/CompactBlockProcessor.swift

```
@@ -367,6 +375,55 @@ private extension CompactBlockProcessor {
     }
 }
 
+// MARK: - Mempool Detection
+
+extension CompactBlockProcessor {
+    func watchMempool() async {
+        while !Task.isCancelled {
+            do {
+                try await consumeMempoolStream()
+                
+                try await Task.sleep(nanoseconds: 500_000_000)
+            } catch {
+                // Connection hammering prevention, wait 30s in case stream failed
+                try? await Task.sleep(nanoseconds: 30_000_000_000)
+            }
+        }
+    }
+    
+    func consumeMempoolStream() async throws {
+        let stream = try service.getMempoolStream()
+        
+        do {
+            for try await rawTransaction in stream {
+                let mempoolTransactions = try await resolveMempools(rawTransaction: rawTransaction)
+                
+                if !mempoolTransactions.isEmpty {
+                    let height = BlockHeight(rawTransaction.height)
+                    await send(event: .foundTransactions(mempoolTransactions, .init(uncheckedBounds: (lower: height, upper: height))))
+                }
+            }
+        } catch { }
```

Wouldn't this ignore any errors occurring in decryptAndStoreTransaction calls inside resolveMempools? is this intended?

---

Since mempool detection is a fully independent new service unrelated to sync loop, there is no follow up what would do anything if `decryptAndStoreTransaction` fails. So it's indeed ignored. Result of `resolveMempools` is either a happy path - the transaction has been detected and broadcasted to client(s) or nothing to report.